### DISCUSSION TITLE
Git commit.template isn't respected?

### DISCUSSION THREAD

**Discussion by IPWright83** [timbrel/GitSavvy#1552] [General]
It could well be I've just missed some configuration but I can't seem to get GitSavvy to respect a `commit.template` file that I've created?

What I want to do is prefix my commit message with a format such as `feat[ticket-id]: `. I've created a pre-commit hook to run to generate this based on a branch name. I've currently set this globally using `git config --global commit.template ~/.gitmessage` and I expected that to be at the start of my commit message, but it always seems to be empty? I've checked in VSCode and it seems to be picking it up fine.

Is there a setting I'm missing or a way to get GitSavvy to respect this? Bonus points if it's possible to actually configure this at the repository level somehow so I don't use the same `~/.gitmessage` on all my repos.

**Comment by kaste** [comment]
We don't run the pre-commit or commit-message hook.  (Don't know the exact names of these hooks; there are plenty of them around committing.)

The plan was to wait for the `git hook` functionality so we don't have to reinvent the wheel.  This feature is WIP since ages now.

**Comment by kaste** [comment]
Related https://github.com/timbrel/GitSavvy/discussions/1538 esp https://github.com/timbrel/GitSavvy/discussions/1538#discussioncomment-2514971

**Reply by IPWright83** [reply]
Ah yeah, that's almost exactly what I want thanks @kaste.
